File: C Sharp/Object Validator/Runtime/Range.cs
using System.Numerics;

namespace Quicksilver.ObjectValidator.Runtime;
class Range(bool reversed, string? fieldExpression, string range) : Condition(reversed, fieldExpression)
{
    private readonly string range = range;

    protected override bool IsFulfilledBy(object? value, string? fullFieldExpression, HashSet<string> passedFields, HashSet<string> failedFields)
    {
        return value switch
        {
            null => throw new Exception("Null values are not supported for 'range'."),
            sbyte or byte or short or ushort or int or uint or long or ulong or BigInteger or float or double or decimal or DateTime => Utils.InRange(value, range),
            _ => throw new Exception("Unsupported type for 'range': " + value.GetType()),
        };
    }
}

File: Java/Object Validator/lib/src/main/java/com/quicksilver/objectvalidator/runtime/Range.java
package com.quicksilver.objectvalidator.runtime;

import java.util.Date;
import java.util.HashSet;

public class Range extends Condition {
    private final String range;

    public Range(boolean reversed, String fieldExpression, String range) {
        super(reversed, fieldExpression);
        this.range = range;
    }

    @Override
    protected boolean isFulfilledBy(Object value, String fullFieldExpression, HashSet<String> passedFields, HashSet<String> failedFields) {
        if (value == null)
            throw new RuntimeException("Null values are not supported for 'range'.");
        if (value instanceof Number || value instanceof Date)
            return Utils.inRange(value, range);
        throw new RuntimeException("Unsupported type for 'range': " + value.getClass());
    }
}
